v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/impl/network/handlers/DenizenNetworkManagerImpl.java

```diff
@@ -90,6 +52,32 @@ public static FriendlyByteBuf copyPacket(Packet<?> original) {
         }
     }
 
+    @FunctionalInterface
+    public interface PacketHandler<T extends Packet<ClientGamePacketListener>> {
+        Packet<ClientGamePacketListener> handlePacket(DenizenNetworkManagerImpl networkManager, T packet) throws Exception;
+    }
+
+    public static final Map<Class<? extends Packet<ClientGamePacketListener>>, List<PacketHandler<?>>> PACKET_HANDLERS = new HashMap<>();
+
+    public static <T extends Packet<ClientGamePacketListener>> void registerPacketHandler(Class<T> packetClass, PacketHandler<T> handler) {
+        PACKET_HANDLERS.computeIfAbsent(packetClass, k -> new ArrayList<>()).add(handler);
+    }
+
+    public static <T extends Packet<ClientGamePacketListener>> void registerPacketHandler(Class<T> packetClass, BiConsumer<DenizenNetworkManagerImpl, T> handler) {
+        registerPacketHandler(packetClass, (networkManager, packet) -> {
+            handler.accept(networkManager, packet);
+            return packet;
+        });
+    }
+
+    public static <T extends Packet<ClientGamePacketListener>> void registerPacketHandlerForChildren(Class<T> parentPacketClass, PacketHandler<T> handler) {
+        for (Class<?> childClass : parentPacketClass.getDeclaredClasses()) {
+            if (parentPacketClass.isAssignableFrom(childClass)) {

```

Comment:
...  huh?

Reply:
See the PR description - this check specifically is incase a packet class has a child class that isn't a packet, but not sure if this entire method is the best way to handle this; main idea was something that won't break if e.g. `MoveEntityPacket.Face` or whatever was added, but might be better off just specifying the classes directly as usual.

Reply:
yeah specifying classes directly seems wiser there

Reply:
Removed that method in favor of directly specifying the classes